### ISSUE TITLE
create way to access runtime parameters from C++

### ISSUE THREAD

**Issue by zingale** [starkiller-astro/Microphysics#245]
We currently parse the `_parameter` files at compile time to learn what parameters are defined.  At runtime, Fortran reads in the parameters to set the value at runtime.  

So we need to have access to the parameters in C++, perhaps as global memory, and then we also need to communicate the values from Fortran to C++ through some init interface.

**Comment by zingale** [comment]
One way of dealing with the parameters is to declare them as `extern` in a header:

`parameters.H`:
```
#ifndef _parameters_H_
#define _parameters_H_

extern int a;
extern double x;

void test_params();

#endif
```

`main.cpp`:
```
#include <iostream>
#include "parameters.H"

int a;
double x;


int main() {

  a = 1.0;

  test_params();

  std::cout << "a = " << a << std::endl;
  std::cout << "x = " << x << std::endl;

  test_params();

  return 0;
}
```

`func.cpp`:
```
#include <iostream>
#include "parameters.H"

void test_params() {

  std::cout << "a = " << a << std::endl;

  a += 1;
  x = 2.0;

}
```

**Comment by zingale** [comment]
another option is to use a simple class and define them as static member variables:

`parameters.H`:
```
#ifndef _parameters_H_
#define _parameters_H_

class Parameters {
  public:

  static int a;
  static double x;
};


void test_params();

#endif
```

`main.cpp`:
```
#include <iostream>
#include "parameters.H"

int Parameters::a;
double Parameters::x;


int main() {

  Parameters::a = 1.0;

  test_params();

  std::cout << "a = " << Parameters::a << std::endl;
  std::cout << "x = " << Parameters::x << std::endl;

  return 0;
}
```

`func.cpp`:
```
#include <iostream>
#include "parameters.H"

void test_params() {

  std::cout << "a = " << Parameters::a << std::endl;

  Parameters::a += 1;
  Parameters::x = 2.0;

}
```

**Comment by zingale** [comment]
this is implemented in #247
